core/store/src/trie/mem/resharding.rs

```diff
@@ -25,6 +32,38 @@ enum RetainDecision {
     Descend,
 }
 
+/// By the boundary account and the retain mode, generates the list of ranges
+/// to be retained in trie.
+fn boundary_account_to_intervals(
+    boundary_account: &AccountId,
+    retain_mode: RetainMode,
+) -> Vec<Range<Vec<u8>>> {
+    let mut intervals = vec![];
+    // TODO(#12074): generate correct intervals in nibbles.
+    for (col, _) in COLUMNS_WITH_ACCOUNT_ID_IN_KEY {
+        match retain_mode {
+            RetainMode::Left => {
+                intervals.push(vec![col]..[&[col], boundary_account.as_bytes()].concat())
+            }
+            RetainMode::Right => {
+                intervals.push([&[col], boundary_account.as_bytes()].concat()..vec![col + 1])
+            }
+        }
+    }
+    intervals
+}
+
+/// Converts the list of ranges in bytes to the list of ranges in nibbles.
+fn intervals_to_nibbles(intervals: &[Range<Vec<u8>>]) -> Vec<Range<Vec<u8>>> {
```

Comment:
nit: maybe newtype for converted list would be nice, so that we know if we deal with already converted list.
But I see you added `// TODO(#12074): generate correct intervals in nibbles.` so I guess the newtype would be removed later anyway.

Reply:
That's fair, but nibbles & bytes are already poorly typed in the whole code. I'd prefer to merge this one first and then consider fixing typing everywhere. There is `NibbleSlice` but I'm a bit scared to use it.